WWDC/SessionsTableViewController.swift

```diff
@@ -261,12 +261,29 @@ class SessionsTableViewController: NSViewController, NSMenuItemValidation {
                     selectedIndexes.insert(overrideIndex)
                 } else {
                     // Preserve selected rows if possible
-                    let selectedRows = self.tableView.selectedRowIndexes.compactMap { (index) -> IndexedSessionRow? in
+                    let previouslySelectedRows = self.tableView.selectedRowIndexes.compactMap { (index) -> IndexedSessionRow? in
                         guard index < oldValue.endIndex else { return nil }
                         return IndexedSessionRow(sessionRow: oldValue[index], index: index)
                     }
 
-                    selectedIndexes = IndexSet(newRowsSet.intersection(selectedRows).map { $0.index })
+                    let newSelection = newRowsSet.intersection(previouslySelectedRows)
+                    if let topOfPreviousSelection = previouslySelectedRows.first, newSelection.isEmpty {
+                        // The update has removed the selected row(s).
+                        // e.g. You have the unwatched filter active and then mark the selection as watched
+                        stride(from: topOfPreviousSelection.index, to: -1, by: -1).lazy.compactMap {
+                            return IndexedSessionRow(sessionRow: oldValue[$0], index: $0)
+                        }.first { (indexedRow: IndexedSessionRow) -> Bool in
```

Comment:
As a result of writing Kotlin for 1.5 years, I feel lost in Swift. But I don't think you can get around all the nulls. I could stack some `if let`s probably